### PR TITLE
Taking into account the PageMode when pull related content

### PR DESCRIPTION
**Pull request by freddyucv**
Taking PageMode into account when pull related content

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/ContentMap.java [hunk 1]
```diff
@@ -342,20 +336,49 @@ private Object getRelationshipInfo(final Field field)
...
+	private Contentlet getLiveVersion(final Contentlet relatedContent) {
+		try {
+			if (relatedContent.isLive()) {
+				return relatedContent;
+			} else {
+				try {
+					final List<Contentlet>  contentlets = APILocator.getContentletAPI()
+							.findContentletsByIdentifiers(new String[]{relatedContent.getIdentifier()}, true, relatedContent.getLanguageId(), user, user.isFrontendUser());
+
+					return contentlets.get(0);
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/ContentMap.java:364]: I would check if it is not null or empty list

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/ContentMap.java [hunk 2]
```diff
@@ -342,20 +336,49 @@ private Object getRelationshipInfo(final Field field)
...
+				} catch (DotDataException| DotSecurityException e) {
+					throw new RuntimeException(e);
+				}
+			}
+		} catch (DotDataException | DotSecurityException e) {
+			return relatedContent;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/ContentMap.java:370]: is this ok

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/ContentMap.java [hunk 3]
```diff
@@ -342,20 +336,49 @@ private Object getRelationshipInfo(final Field field)
...
+		}
+	}
+
+	private boolean needLiveVersion(final Contentlet contentlet) {
+		try {
+			return EDIT_OR_PREVIEW_MODE || !EDIT_OR_PREVIEW_MODE && contentlet.hasLiveVersion();
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/ContentMap.java:376]: you ca use Try.of

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/ContentMap.java (hunk 1 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/ContentMap.java]: done https://github.com/dotCMS/core/pull/19551/commits/9ed56d60337cd0089bc8c6b1562ecf5fcd027a50#diff-fbb11290830742365067c6c4e761a23f3eef90d1e024037cbaa272100c134e61R365

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/ContentMap.java (hunk 2 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/ContentMap.java]: done https://github.com/dotCMS/core/pull/19551/commits/9ed56d60337cd0089bc8c6b1562ecf5fcd027a50#diff-fbb11290830742365067c6c4e761a23f3eef90d1e024037cbaa272100c134e61R371

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/ContentMap.java (hunk 3 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/ContentMap.java]: done https://github.com/dotCMS/core/pull/19551/commits/9ed56d60337cd0089bc8c6b1562ecf5fcd027a50#diff-fbb11290830742365067c6c4e761a23f3eef90d1e024037cbaa272100c134e61R376

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/ContentMap.java [hunk 4]
```diff
@@ -342,20 +336,47 @@ private Object getRelationshipInfo(final Field field)
...
+					final List<Contentlet> contentlets = APILocator.getContentletAPI()
+							.findContentletsByIdentifiers(new String[]{relatedContent.getIdentifier()}, true, relatedContent.getLanguageId(), user, user.isFrontendUser());
+
+					return contentlets != null && !contentlets.isEmpty() ? contentlets.get(0) : null;
+				} catch (DotDataException| DotSecurityException e) {
+					throw new RuntimeException(e);
+				}
+			}
+		} catch (DotDataException | DotSecurityException e) {
+			return null;
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/ContentMap.java:371]: Is it better to return Optional?

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/ContentMap.java [hunk 5]
```diff
@@ -342,20 +336,47 @@ private Object getRelationshipInfo(final Field field)
...
+	private Contentlet getLiveVersion(final Contentlet relatedContent) {
+		try {
+			if (relatedContent.isLive()) {
+				return relatedContent;
+			} else {
+				try {
+					final List<Contentlet> contentlets = APILocator.getContentletAPI()
+							.findContentletsByIdentifiers(new String[]{relatedContent.getIdentifier()}, true, relatedContent.getLanguageId(), user, user.isFrontendUser());
+
+					return contentlets != null && !contentlets.isEmpty() ? contentlets.get(0) : null;
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/ContentMap.java:365]: Return an empty list rather than null.

**Review by jdotcms — APPROVED** [review]

**Pull request merged**